tools/smithy-rs-sync/src/main.rs

```diff
@@ -0,0 +1,409 @@
+    if !remove_examples_command_output.status.success() {
+        bail!(
+            "failed to clean out the examples folder at {}",
+            examples_path
+        );
+    }
+
+    println!("\tsuccessfully cleaned out previously built SDK");
+    Ok(())
+}
+
+/// Use `cp -r` to recursively copy all files and folders from the smithy-rs build artifacts folder
+/// to the aws-sdk-rust repo folder
+fn copy_sdk(from_path: &Path, to_path: &Path) -> Result<()> {
+    println!("\tcopying built SDK...");
+
+    let copy_sdk_command_output = Command::new("cp")
+        .arg("-r")
+        .arg(&from_path)
+        .arg(&to_path)
+        .output()?;
+    if !copy_sdk_command_output.status.success() {
+        bail!(
+            "failed to copy the built SDK from {} to {}",
+            from_path.to_string_lossy(),
+            to_path.to_string_lossy()
+        );
+    }
+
+    println!("\tsuccessfully copied built SDK");
+    Ok(())
+}
+
+/// Find the last commit made to a repo
+fn find_last_commit(repo: &Repository) -> Result<Commit> {
+    let obj = repo
+        .head()
+        .context(here!())?
+        .resolve()
+        .context(here!())?
+        .peel(ObjectType::Commit)
+        .context(here!())?;
+    obj.into_commit()
+        .map_err(|_| anyhow!("couldn't find last commit"))
+}
+
+/// Create a "mirror" commit. Works by reading a smithy-rs commit and then using the info
+/// attached to it to create a commit in aws-sdk-rust.
+fn create_mirror_commit(aws_sdk_repo: &Repository, based_on_commit: &Commit) -> Result<()> {
+    println!("\tcreating mirror commit...");
+
+    let mut index = aws_sdk_repo.index().context(here!())?;
+    // The equivalent of `git add .`
+    index
+        .add_all(["."].iter(), IndexAddOption::DEFAULT, None)
+        .context(here!())?;
+    let oid = index.write_tree().context(here!())?;
+    let parent_commit = find_last_commit(&aws_sdk_repo).context(here!())?;
+    let tree = aws_sdk_repo.find_tree(oid).context(here!())?;
+
+    let _ = aws_sdk_repo
+        .commit(
+            Some("HEAD"),
+            &based_on_commit.author(),
+            &based_on_commit.committer(),
+            based_on_commit.message().unwrap_or_default(),
+            &tree,
+            &[&parent_commit],
+        )
+        .context(here!())?;
+
+    println!("\tsuccessfully created mirror commit");
+
+    Ok(())
+}
+
+/// Commit the file in aws-sdk-rust that tracks what smithy-rs commit the SDK was last built from
+fn commit_last_synced_commit_file(aws_sdk_repo: &Repository) -> Result<()> {
+    let mut index = aws_sdk_repo.index().context(here!())?;
+    index
+        .add_path(Path::new(COMMIT_HASH_FILENAME))
+        .context(here!())?;
+    let signature = Signature::now(BOT_NAME, BOT_EMAIL)?;
+    let oid = index.write_tree().context(here!())?;
+    let parent_commit = find_last_commit(&aws_sdk_repo).context(here!())?;
+    let tree = aws_sdk_repo.find_tree(oid).context(here!())?;
+
+    let _ = aws_sdk_repo
```

Comment:
```suggestion
    aws_sdk_repo
```

Reply:
The commit method being called on the next line returns an Oid. I'm assigning to _ to be explicit about throwing away that value.